common/libs/transaction.js

```diff
@@ -29,3 +39,95 @@ export type BroadcastTransaction = {|
   rawTx: string,
   signedTx: string
 |};
+
+export async function generateTransaction(
+  node: BaseNode,
+  tx: Transaction,
+  wallet: BaseWallet,
+  token: ?Token
```

Comment:
If you want to move token along everywhere it should be inside transaction object